src/WebJobs.Script.WebHost/Security/SecretManager.cs

```diff
@@ -335,7 +337,23 @@ private Task<ScriptSecrets> LoadSecretsAsync(ScriptSecretsType secretsType, stri
 
         private async Task<ScriptSecrets> LoadSecretsAsync(ScriptSecretsType type, string keyScope, Func<string, ScriptSecrets> deserializationHandler)
         {
-            string secretsJson = await _repository.ReadAsync(type, keyScope).ConfigureAwait(false);
+            string secretsJson;
+            for (int currentRetry = 0; ; currentRetry++)
+            {
+                try
+                {
+                    secretsJson = await _repository.ReadAsync(type, keyScope).ConfigureAwait(false);
+                    break;
+                }
+                catch (IOException)
```

Comment:
This class generalizes the storage repository (e.g. file based and blob based repositories). However, IOException is file specific? I don't think we should have repository specific logic in this general class.

Is the issue only happening for file storage? Perhaps we should push this file based retry logic further down into the FileUtility?

Reply:
The idea was to make this change on SecurityManager level so we'll have retry logic for both (FileSystem and BlobStorage). Then we can just add new exception type we want to catch for retry.

Reply:
I agree. As previously discussed, we want to keep any notion of file access/IO outside of this class as it should make not assumption about how the secrets are persisted (or if they are persisted).